spec/unit/semver_spec.rb

```diff
@@ -212,6 +212,17 @@
       expect(version.tiny).to    eq(5)
       expect(version.special).to eq('-beta6')
     end
+
+    it 'should not log a deprecation warning when strict == off' do
+      Puppet[:strict] = :off
+      SemVer.new('1.2.3')
+      expect(@logs).to be_empty
+    end
+
+    it 'should log a deprecation warning unless strict == off' do
```

Comment:
True, on the other hand, there's no code introduced that is specific warning and error.
Neither the spec, the code, nor the test is concerned about warning or error. It strictly tests if strict is off, then issue no warning. If it isn't then warn. I fail to see what the problem is here.